HopsanGUI/Widgets/ModelWidget.cpp

```diff
@@ -698,16 +698,260 @@ void ModelWidget::saveAs()
+        if(usedUnits.contains(qMakePair(units[i], quantity)) || quantity.isEmpty()) {
+            continue;
+        }
+        usedUnits.append(qMakePair(units[i], quantity));
+        QString baseUnit = gpConfig->getBaseUnit(quantity);
+        UnitConverter converter;
+        gpConfig->getUnitScale(quantity, units[i], converter);
+        double scale = converter.scaleToDouble();
+        double offset = converter.offsetToDouble();
+        QDomElement ssvUnitElement = appendDomElement(ssvUnitsElement, "ssv:Unit");
+        ssvUnitElement.setAttribute("name", units[i]);
+
+        QDomElement ssvBaseUnitElement = appendDomElement(ssvUnitElement, "ssv:BaseUnit");
+        ssvBaseUnitElement.setAttribute("factor", scale);
+        ssvBaseUnitElement.setAttribute("offset", offset);
+
+        //! @todo We should maybe derive base units in this way everywhere to conform with e.g. FMI and SSP standards, instead of hard-coding it here
+        int kg, m, s, A, K, mol, cd, rad;
+        gpConfig->getBaseUnitSIExponents(quantity, kg, m, s, A, K, mol, cd, rad);
+        ssvBaseUnitElement.setAttribute("kg",  kg);
+        ssvBaseUnitElement.setAttribute("m",   m);
+        ssvBaseUnitElement.setAttribute("s",   s);
+        ssvBaseUnitElement.setAttribute("A",   A);
+        ssvBaseUnitElement.setAttribute("K",   K);
+        ssvBaseUnitElement.setAttribute("mol", mol);
+        ssvBaseUnitElement.setAttribute("cd",  cd);
+        ssvBaseUnitElement.setAttribute("rad", rad);
+    }
+
+    QFile file;
+    file.setFileName(ssvFile);   //Create a QFile object
+    if (!file.open(QIODevice::WriteOnly | QIODevice::Text)) {
+        gpMessageHandler->addErrorMessage("Failed to open file for writing: " + ssvFile);
+        return;
+    }
+    QTextStream fileStream(&file);
+    ssvDocument.save(fileStream, XMLINDENTATION);
+    file.close();
+}
+
+void ModelWidget::importModelParametersFromHpf(QString parameterFile)
 {
     mpToplevelSystem->loadParameterValuesFromFile(parameterFile);
 }
 
+void ModelWidget::importModelParametersFromSsv()
+{
+    //Let user choose SSV file
+    QString openLocation = gpConfig->getStringSetting(CFG_PARAMETERIMPORTDIR);
+    if (openLocation.isEmpty()) {
+        openLocation = gpConfig->getStringSetting(CFG_LOADMODELDIR);
+    }
+    QString parameterFile = QFileDialog::getOpenFileName(gpMainWindowWidget, tr("Load Parameters from SSV File"),
+                                                 openLocation,
+                                                 tr("System Structure Parameter Values (*.ssv)"));
+    if(!parameterFile.isEmpty()) {
+        gpConfig->setStringSetting(CFG_PARAMETERIMPORTDIR,  QFileInfo(parameterFile).absolutePath());
+    }
+
+    //This map translates data type names from Hopsan to SSV
+    QMap<QString, QString> dataTypeTranslator{{"double", "Real"}, {"integer", "Integer"}, {"bool", "Boolean"}, {"string", "String"}, {"textblock", "String"}, {"filepath", "Strig"}, {"conditional", "Integer"}};
```

Comment:
Again {"filepath", "Strig"}
Maybe this translation map could be one global map variable in an anonymous namespace (int this file). But not sure if you can create QMap like that. But it would be better to have one, instead of two copies. Maybe a std::uniqu_ptr or qsharedpointer could be used. Or maybe its simpler this way.

Reply:
I moved the map to `XMLUtilities.h` and made it static (as part of the `ssv`  namesapce)